Podcast/ListeningHistoryEndpointRequests.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  ListeningHistoryEndpointRequests.swift
+//  Podcast
+//
+//  Created by Jack Thompson on 8/27/18.
+//  Copyright © 2018 Cornell App Development. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+class FetchListeningHistoryEndpointRequest: EndpointRequest {
+    
+    var offset: Int
+    var max: Int
+    
+    // dismissed is an optional argument that if supplied will filter the
+    // listening history such that all returned listening history entries have their dismissed value equal to the supplied value.
+    var dismissed: Bool?
+    
+    init(offset: Int, max: Int, dismissed: Bool? = nil) {
+        self.offset = offset
+        self.max = max
+        self.dismissed = dismissed
+        super.init()
+        
+        path = "/history/listening/"
+        httpMethod = .get
+        queryParameters = ["offset": offset, "max": max]
+        if let dismiss = dismissed {
+            queryParameters!["dismissed"] = encodeBoolean(dismiss)
```

Comment:
same here

Reply:
potentially make encodeBoolean an extension of Bool so you can do `dismiss.encode()`